ios/Runner/AppDelegate.swift

```diff
@@ -3,11 +3,72 @@ import Flutter
 
 @main
 @objc class AppDelegate: FlutterAppDelegate {
+  private var notificationTapEventListener: NotificationTapEventListener?
+
   override func application(
     _ application: UIApplication,
     didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
   ) -> Bool {
     GeneratedPluginRegistrant.register(with: self)
+    guard let controller = window?.rootViewController as? FlutterViewController else {
+      fatalError("rootViewController is not type FlutterViewController")
+    }
+
+    // Retrieve the remote notification payload from launch options;
+    // this will be null if the launch wasn't triggered by a notification.
+    let notificationPayload = launchOptions?[.remoteNotification] as? [AnyHashable : Any]
+    let api = NotificationHostApiImpl(notificationPayload.map { NotificationDataFromLaunch(payload: $0) })
+    NotificationHostApiSetup.setUp(binaryMessenger: controller.binaryMessenger, api: api)
+
+    notificationTapEventListener = NotificationTapEventListener()
+    NotificationTapEventsStreamHandler.register(with: controller.binaryMessenger, streamHandler: notificationTapEventListener!)
+
+    // Setup handler for notification tap while the app is running.
+    UNUserNotificationCenter.current().delegate = self
+
     return super.application(application, didFinishLaunchingWithOptions: launchOptions)
   }
+
+  override func userNotificationCenter(
+    _ center: UNUserNotificationCenter,
+    didReceive response: UNNotificationResponse,
+    withCompletionHandler completionHandler: @escaping () -> Void
+  ) {
+    if let listener = notificationTapEventListener {
+      let userInfo = response.notification.request.content.userInfo
+      listener.onNotificationTapEvent(event: NotificationTapEvent(payload: userInfo))
```

Comment:
I think this API would be a bit cleaner if we adjust it like so:
```suggestion
      listener.onNotificationTapEvent(payload: userInfo)
```

Then the NotificationTapEvent gets constructed by the method implementation.

That's also consistent with the Pigeon docs' [example](https://github.com/flutter/packages/blob/e6e814c1ee1894efff93a17293c0d5e3169db05d/packages/pigeon/example/README.md#swift-2):
```swift
  func onIntEvent(event: Int64) {
    if let eventSink = eventSink {
      eventSink.success(IntEvent(data: event))
    }
  }

  func onStringEvent(event: String) {
    if let eventSink = eventSink {
      eventSink.success(StringEvent(data: event))
    }
  }
```